internal/tfdiags/consolidate_warnings.go

```diff
@@ -22,6 +22,11 @@ import "fmt"
 //
 // The definition of "unreasonable" is given as the threshold argument. At most
 // that many diagnostics with the same summary will be shown.
+
+type Keyable interface {
+	ExtraInfoKey() string
+}
+
```

Comment:
Please move the interface from here since the comment above is about the `Consolidate` method.